### ISSUE TITLE
TouchableWithoutFeedback Component doesn't disable click functionality when disabled

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30953]
## Description
When using a screen reader the TouchableWithoutFeedback component doesn't disable click functionality.

This issue is covered in-depth by parent issue #30840, please check there for more details.

## React Native version:
0.63

**Comment by saurabhkacholiya** [comment]
Hi, @amarlette I am new to opensource can I pick up this issue?

**Comment by amarlette** [comment]
Yes you can! I would recommend checking out our contribution guide (https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md) to help get you started.

**Comment by saurabhkacholiya** [comment]
I have tested TouchableWithoutFeedback is working fine this commit [Here](https://github.com/facebook/react-native/issues/30949#issuecomment-777067982) solve the issue

**Comment by amarlette** [comment]
@saurabhkacholiya can you attach the test that you used to confirm this?

**Comment by fabriziobertoglio1987** [comment]
>I have tested TouchableWithoutFeedback is working fine this commit Here solve the issue

I would say no

https://github.com/facebook/react-native/blob/1c7d9c8046099eab8db4a460bedc0b2c07ed06df/Libraries/Components/Pressable/Pressable.js#L142
https://github.com/facebook/react-native/blob/1c7d9c8046099eab8db4a460bedc0b2c07ed06df/Libraries/Components/Pressable/Pressable.js#L171-L174

those changes are in the `Pressable` component and I could not find any usage of `Pressable` in `TouchableWithoutFeedback`.
This makes sense as `TouchableWithoutFeedback` was built before `Pressable`, but I don't know well those `.js` libraries... 
Tomorrow I will further investigate and try to add the props in 

https://github.com/facebook/react-native/blob/e04fa92bc0d0f829de2f4dab7b4c244bf8b5d889/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L118

:peace_symbol:  :pray:

**Comment by saurabhkacholiya** [comment]
@amarlette I am adding snapshots and test cases please validate.
Test Cases
```
describe('<TouchableWithoutFeedback />', () => {
  it('should render as expected', () => {
    const instance = ReactTestRenderer.create(
      <TouchableWithoutFeedback>
        <Text>Touchable</Text>
      </TouchableWithoutFeedback>,
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});

describe('<TouchableWithoutFeedback disabled={true} />', () => {
  it('should be disabled when disabled is true', () => {
    const instance = ReactTestRenderer.create(
      <TouchableWithoutFeedback disabled={true}>
        <Text>Touchable</Text>
      </TouchableWithoutFeedback>,
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});

describe('<TouchableWithoutFeedback disabled={true} accessibilityState={{}} />', () => {
  it('should be disabled when disabled is true and accessibilityState is empty', () => {
    const instance = ReactTestRenderer.create(
      <TouchableWithoutFeedback disabled={true} accessibilityState={{}}>
        <Text>Touchable</Text>
      </TouchableWithoutFeedback>,
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});

describe('<TouchableWithoutFeedback disabled={true} accessibilityState={{checked: true}} />', () => {
  it('should keep accessibilityState when disabled is true', () => {
    const instance = ReactTestRenderer.create(
      <TouchableWithoutFeedback
        disabled={true}
        accessibilityState={{checked: true}}>
        <Text>Touchable</Text>
      </TouchableWithoutFeedback>,
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});

describe('<TouchableWithoutFeedback disabled={true} accessibilityState={{disabled: false}} />', () => {
  it('should overwrite accessibilityState with value of disabled prop', () => {
    const instance = ReactTestRenderer.create(
      <TouchableWithoutFeedback
        disabled={true}
        accessibilityState={{disabled: false}}>
        <Text>Touchable</Text>
      </TouchableWithoutFeedback>,
    );
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
```
SnapShot
```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TouchableWithoutFeedback /> should render as expected 1`] = `
<Text
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback /> should render as expected: should deep render when mocked (please verify output manually) 1`] = `
<Text
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback /> should render as expected: should deep render when not mocked (please verify output manually) 1`] = `
<Text
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback /> should render as expected: should shallow render as <TouchableWithoutFeedback /> when mocked 1`] = `
<TouchableWithoutFeedback>
  <Text>
    Touchable
  </Text>
</TouchableWithoutFeedback>
`;

exports[`<TouchableWithoutFeedback /> should render as expected: should shallow render as <TouchableWithoutFeedback /> when not mocked 1`] = `
<TouchableWithoutFeedback>
  <Text>
    Touchable
  </Text>
</TouchableWithoutFeedback>
`;

exports[`<TouchableWithoutFeedback disabled={true} /> should be disabled when disabled is true 1`] = `
<Text
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty 1`] = `
<Text
  accessibilityState={Object {}}
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true 1`] = `
<Text
  accessibilityState={
    Object {
      "checked": true,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop 1`] = `
<Text
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`TouchableHighlight renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
>
  <Text>
    Touchable
  </Text>
</View>
`;

```

**Comment by saurabhkacholiya** [comment]
@fabriziobertoglio1987 Pressability is being used in TouchableWithoutFeedback
> https://github.com/facebook/react-native/blob/3ff7e86b0ff12823e5d89a0d3a7cceb79e0b25b0/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L11-L13
> https://github.com/facebook/react-native/blob/3ff7e86b0ff12823e5d89a0d3a7cceb79e0b25b0/Libraries/Components/Touchable/TouchableWithoutFeedback.js#L92-L95

**Comment by fabriziobertoglio1987** [comment]
thanks @saurabhkacholiya but sorry, I think you are wrong. 
I want to prepare a Pull Request for this issue tomorrow.

Your snapshot https://github.com/facebook/react-native/issues/30953#issuecomment-800524578 is wrong `accessibilityState={ disabled: false}`

>```
>exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState >with value of disabled prop 1`] = `
><Text
>  accessibilityState={
>    Object {
>      "disabled": false,
>    }
>  }
>```

while the commit https://github.com/facebook/react-native/commit/1c7d9c8046099eab8db4a460bedc0b2c07ed06df has the correct snapshot with `accessibilityState={ disabled: true}`

https://github.com/facebook/react-native/blob/1c7d9c8046099eab8db4a460bedc0b2c07ed06df/Libraries/Components/Pressable/__tests__/__snapshots__/Pressable-test.js.snap#L283-L289

Facebook Team requested us to build this feature in accessability and change `accessabilityState.disabled` based on `disabled` prop.

**Comment by fabriziobertoglio1987** [comment]
@saurabhkacholiya as you have already wrote test for this fix, you can check my solution, test it and verify it actually works and then publish a pull request with your github account https://github.com/fabriziobertoglio1987/react-native/commit/153a37ce5cf8464c9f951ffd64d7641954fe8660

thanks a lot. Bye

**Comment by saurabhkacholiya** [comment]
@fabriziobertoglio1987 Sure I would like to give a pull request Thanks a lot. can you just validate the snapshot 
```
exports[`<TouchableWithoutFeedback /> should render as expected 1`] = `
<Text
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} /> should be disabled when disabled is true 1`] = `
<Text
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{}} /> should be disabled when disabled is true and accessibilityState is empty 1`] = `
<Text
  accessibilityState={Object {}}
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{checked: true}} /> should keep accessibilityState when disabled is true 1`] = `
<Text
  accessibilityState={
    Object {
      "checked": true,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;

exports[`<TouchableWithoutFeedback disabled={true} accessibilityState={{disabled: false}} /> should overwrite accessibilityState with value of disabled prop 1`] = `
<Text
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  Touchable
</Text>
`;
```

**Comment by fabriziobertoglio1987** [comment]
I notice just now 

https://github.com/facebook/react-native/commit/1c7d9c8046099eab8db4a460bedc0b2c07ed06df#diff-466095fa96dd3e4ad5dffc7f31d8da5084c8a0acd3396a8b65b45cd99c48dd4bR320-R330

when you run the test in shallow render, the test output is not correct .. but seems to be ok for the Facebook team ... so I would not worry about that

The important thing is that they work when deep rendering the output 

https://github.com/facebook/react-native/commit/1c7d9c8046099eab8db4a460bedc0b2c07ed06df#diff-466095fa96dd3e4ad5dffc7f31d8da5084c8a0acd3396a8b65b45cd99c48dd4bR283-R304

**Comment by amarlette** [comment]
@carloscuesta thank you for opening up a PR for this issue!
